kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -1249,6 +1256,35 @@ recursiveForcedStep n node
         SingleResult sr -> (recursiveForcedStep $ n-1) sr
         BranchResult xs -> Foldable.traverse_ (recursiveForcedStep (n-1)) xs
 
+{- | Returns the first interesting branching node encountered by
+exploring the proof graph for 'n' steps over all branches, starting
+from 'node'. If no such node exists, it tries to return the only existing
+non-bottom leaf. If no such leaf exists, it returns Nothing.
+-}
+
+getInterestingBranchingNode
+    :: Natural
+    -> InnerGraph axiom
+    -> ReplNode
+    -> Maybe ReplNode
+getInterestingBranchingNode n graph node
+    | nodeIsBottom               = Nothing
+    | n == 0                     = Just node
+    | null sucResults            = Just node
+    | otherwise =
+        case interestingResults of
+            []  -> Nothing
+            [a] -> Just a
+            _   -> Just node
+    where
+      gnode = unReplNode node
+      nodeIsBottom = (==) (getNodeState graph gnode) Nothing
+      sucResults = fmap (getInterestingBranchingNode (n - 1) graph . ReplNode)
+                        (Graph.suc graph gnode)
```

Comment:
Please change the indentation here. See https://github.com/kframework/kore/wiki/Coding-Standards#line-splitting